chrome/content/zotero/xpcom/data/notes.js

```diff
@@ -96,7 +96,36 @@ Zotero.Notes = new function() {
 			await Zotero.Notifier.trigger('refresh', 'item', idsToRefresh);
 		});
 	};
-	
+
+	/**
+	 * Update item key URLs in the item's note, replacing all instances of each
+	 * key in itemKeyMap with the associated value.
+	 * Passed item should have an embedded note or be a note item.
+	 *
+	 * @param {Zotero.Item} item
+	 * @param {Map<String, String>} itemKeyMap
+	 */
+	this.replaceAllItemKeys = function (item, itemKeyMap) {
+		let note = item.getNote();
+		let keys = [...itemKeyMap.keys()].join('|');
+		let re = new RegExp(`%2Fitems%2F(${keys})`, 'g');
+		note = note.replace(re, (str, key) => `%2Fitems%2F${itemKeyMap.get(key)}`);
+		re = new RegExp(`data-attachment-key="${keys}"`);
```

Comment:
Missing parentheses, so this would be matching, e.g., `data-attachment-key="foo` or `bar"`

And let's add a test with at least three keys so we would've caught this.